packages/compiler/core/parser.ts

```diff
@@ -1283,6 +1276,7 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
       kind: SyntaxKind.Projection,
       body,
       direction,
+      directionId: directionId as IdentifierNode<"to" | "from">,
```

Comment:
I was think we wouldn't keep the  IdentifierNode<T>. The directionId could just be an IdentifierNode, no? I don't think we want to lie in the typing here.

Reply:
well we don't lie in the typing, it can still only be those but I guess it doesn't have real use now aprt from being just more precise typing

Reply:
Really? I'm missing how sv is guaranteed to be "to" | "from" above if we took the if with the error above.

Reply:
ah yeah nvm, it is always an identifier node just with the wrong values if there is an error. I'll update